### ISSUE TITLE
Cordova InAppBrowser support broken

### ISSUE THREAD

**Issue by dobesv** [auth0/auth0.js#331]
The v7 branch has special code to support cordova / electron apps and uses different logic for opening the login popups.  It seems like this has been lost in the v8 branch.

In the v8 branch the login popup is opened using the system browser instead of the InAppBrowser.  Newer versions of auth0.js are using "winchan" to open the browser window.  In order for auth0 to work in the InAppBrowser, the second parameter of window.open() must be "_blank".  However, auth0 is now passing `null` for that parameter instead (via winchan).  In theory auth0 could pass `window_name: "_blank"` as an option to winchan if in a cordova environment.

A workaround for this is to monkeypatch `window.open` to force the target to `_blank` and call the InAppBrowser, e.g.:

```
<script>
if(cordova) {
    window.open = function(url, target, opts, replace) {
        return cordova.InAppBrowser.open(url, '_blank', opts, replace);
    };
}
</script>
```

This doesn't get the Auth0Lock working on cordova, but it seems to get me a bit closer.

**Comment by glena** [comment]
hi @dobesv we are working on a new cordova sample which properly handle this by showing the hosted login page in the web browser (not the inapp browser). This way SSO is supported correctly, it is oauth2 compliant and it is also compliant with all the new requirements for social login (not showing the social login page in the inapp browser).

Anyway, we will see if it is worth to bake in lock this workaround. This library is meant for the browser and we are trying to keep it as clean as possible after the rewrite (v8).

**Comment by dobesv** [comment]
OK I will keep using an old version of lock until you have figured this out.

**Comment by meticoeus** [comment]
@glena Apple has been rejecting apps that use the web browser rather than the InAppBrowser to login so this seems like the wrong to direction to go without at least making the browser selection configurable.

[#883](https://github.com/auth0/lock/issues/883)

**Comment by hzalaz** [comment]
@meticoeus thats a different issue that uses a feature that was not meant to be used in cordova apps. Also the InAppBrowser is something that is not recommended for login since some Identity Providers like Google are blocking them. We are working on a sample to first use an in-app browser to show Lock and then we'll see how we'll improve it.

Also the latest release might fix the issues we had with cordova besides #883
